From f3da4f7a741acb2fc8ddc767f448580d32b1b8b2 Mon Sep 17 00:00:00 2001
From: Chanh Nguyen <chanh@os.amperecomputing.com>
Date: Sun, 3 Nov 2024 10:46:44 +0000
Subject: [PATCH] Don't delete DefaultGateway when disabling DHCP

After adding a static IPv4 address, if patch the DHCP disabling,
then DefaultGateway is deleted. That makes the static IPv4 address
fail.

This patch is a temporary fix. The issue was opened a ticket to
github [1]. I will update the better solution after discussing it
with community maintainers.

[1] https://github.com/openbmc/bmcweb/issues/294

Fixes: https://github.com/openbmc/bmcweb/commit/743eb1c0cb44c955078daa2912767cd6d6bb6432
Change-Id: I813d620ea672a66a9dfccb00fd8f42c0e9b3216c
Signed-off-by: Chanh Nguyen <chanh@os.amperecomputing.com>
---
 redfish-core/lib/ethernet.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index a0dcd543..656ee446 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -1385,7 +1385,7 @@ inline void handleDHCPPatch(
 
     if (ipv4Active)
     {
-        updateIPv4DefaultGateway(ifaceId, "", asyncResp);
+        // updateIPv4DefaultGateway(ifaceId, "", asyncResp);
     }
     bool nextv4DHCPState =
         v4dhcpParms.dhcpv4Enabled ? *v4dhcpParms.dhcpv4Enabled : ipv4Active;
-- 
2.43.0

